### PR TITLE
Mods to CMakeLists.txt to support CASA building.

### PR DESCRIPTION
**Pull request by JimNrao**
CASA uses a somewhat unique numbering scheme for it's shared libraries.  In this cmake file if the user sets CASA_BUILD=1 then the build logic will use CASA's numbering scheme; when that setting is not utilized, only the standard cmake logic will be invoked.  

I'm making this pull request at the request of our build guy who's out of the country on an extended vacation (lucky bastard).  The problem we have is that we build and semi-release lots of versions of the various shared libraries so we need a scheme to help manage them.

### CODE REVIEW

**Comment by JimNrao** [comment]
This should be the last mod we need to be able to build CASA against the current version from github so it's important that we get it in as quickly as is reasonable.  Once this is accomplished we should be able to get out automatic build and test system to run using the code from github.

File: CMakeLists.txt [hunk 1]
```diff
@@ -97,18 +116,18 @@ endif (NOT CMAKE_CXX_FLAGS)
 if (NOT CMAKE_BUILD_TYPE)
     # Use debug mode if building in dbg or debug directory.
     get_filename_component(_cmpvar ${CMAKE_BINARY_DIR} NAME)
-    if(_cmpvar STREQUAL "dbg" OR _cmpvar STREQUAL "debug")
+    if("_cmpvar" STREQUAL "dbg" OR "_cmpvar" STREQUAL "debug")
```
Comment (juliantaylor) [inline CMakeLists.txt]: these quotes where removed in gh-110, are they needed again or is this unintended?

Comment (JimNrao) [inline CMakeLists.txt]: Unintended.  Either screwed up the merge or failed to get the latest 
copy.  I did

>  1019  git pull upstream master
>  1021  git checkout -b LibrarySoVersioning

to make the feature branch starting with the latest from github 
(upstream is casacore/casacore); at least that was my intent.


On 07/01/2015 01:39 PM, Julian Taylor wrote:
>
> In CMakeLists.txt 
> <https://github.com/casacore/casacore/pull/144#discussion_r33716153>:
>
> > @@ -97,18 +116,18 @@ endif (NOT CMAKE_CXX_FLAGS)
> >  if (NOT CMAKE_BUILD_TYPE)
> >      # Use debug mode if building in dbg or debug directory.
> >      get_filename_component(_cmpvar ${CMAKE_BINARY_DIR} NAME)
> > -    if(_cmpvar STREQUAL "dbg" OR _cmpvar STREQUAL "debug")
> > +    if("_cmpvar" STREQUAL "dbg" OR "_cmpvar" STREQUAL "debug")
>
> these quotes where removed in gh-110 
> <https://github.com/casacore/casacore/pull/110>, are they needed again 
> or is this unintended?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/casacore/casacore/pull/144/files#r33716153>.Web 
> Bug from 
> https://github.com/notifications/beacon/AGPGtOvYyUV3Wn9xnODg0y9HXdbbdZ-2ks5oZDlugaJpZM4FQHXV.gif
>

-- 
Jim Jacobs

phone: 835-7235
office: 301

Comment (juliantaylor) [inline CMakeLists.txt]: not sure what went wrong, should have worked
but you can quickly fix it via e.g.

    # unstage last commit
    git reset HEAD^
    # interactive stage stuff back
    git add -p
    # commit force push

you will have to write the commit message again (can be copied from git reflog if you forgot)

**Comment by tammojan** [comment]
If you want I can merge it with the quotes. Building with Casa is more important and we can remove the quotes again. 

(A more proper way, if you feel like learning git: remove the quotes, commit, then rebase to squash the two commits.)

**Pull request closed without merging**